sql/src/main/java/io/crate/expression/scalar/SubscriptObjectFunction.java

```diff
@@ -26,51 +26,55 @@
 import io.crate.expression.symbol.Function;
 import io.crate.expression.symbol.Literal;
 import io.crate.expression.symbol.Symbol;
-import io.crate.metadata.BaseFunctionResolver;
 import io.crate.metadata.FunctionIdent;
-import io.crate.metadata.FunctionImplementation;
 import io.crate.metadata.FunctionInfo;
 import io.crate.metadata.Scalar;
 import io.crate.metadata.TransactionContext;
-import io.crate.metadata.functions.params.FuncParams;
-import io.crate.metadata.functions.params.Param;
+import io.crate.metadata.functions.Signature;
 import io.crate.types.DataType;
 import io.crate.types.DataTypes;
 import io.crate.types.ObjectType;
-import io.crate.types.StringType;
 
 import javax.annotation.Nullable;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
+import static io.crate.metadata.functions.TypeVariableConstraint.typeVariable;
+
 /**
  * Scalar function to resolve elements inside a map.
  */
 public class SubscriptObjectFunction extends Scalar<Object, Map<String, Object>> {
 
     public static final String NAME = "subscript_obj";
-    private static final FuncParams FUNCTION_PARAMS = FuncParams
-        .builder(Param.of(ObjectType.untyped()), Param.of(StringType.INSTANCE))
-        .withVarArgs(Param.of(StringType.INSTANCE))
-        .build();
 
-    private FunctionInfo info;
+    public static final Signature SIGNATURE = Signature
+        .scalar(
+            NAME,
+            ObjectType.untyped().getTypeSignature(),
+            DataTypes.STRING.getTypeSignature(),
+            DataTypes.UNDEFINED.getTypeSignature())
+        .withVariableArity()
+        .withTypeVariableConstraints(typeVariable("text"));
```

Comment:
There are no type variable constraints at the arguments or return types. what is this for?

Reply:
as discussed, i'll, of course, remove it. it is just some impl artifact..